Account/FxADeviceRegistration.swift

```diff
@@ -54,7 +54,7 @@ open class FxADeviceRegistration: NSObject, NSCoding {
 
     public convenience required init(coder: NSCoder) {
         let id = coder.decodeObject(forKey: "id") as! String
-        let version = coder.decodeObject(forKey: "version") as! Int
+        let version: Int = coder.decodeObject(forKey: "version") as! Int? ?? 0
```

Comment:
It seems weird that this would be causing problems because of the migration. I'm a bit concerned that by defaulting to 0 we might be breaking some assumptions with the device registration stuff. Could it be because in Swift 2.3 this was a failable initializer but now in Swift 3 it's not?